packages/cadl-ranch-specs/http/payload/pageable/main.tsp

```diff
@@ -0,0 +1,64 @@
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-azure-core";
+import "@typespec/versioning";
+
+using TypeSpec.Versioning;
+using Azure.Core;
+
+@doc("Test describing pageable.")
+@supportedBy("dpg")
+@scenarioService("/payload/pageable")
+@useDependency(global.Azure.Core.Versions.v1_0_Preview_2)
+namespace Payload.Pageable;
+
+@doc("User model")
+model User {
+  @doc("User name")
+  name: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing pageable"
+@scenario
+@scenarioDoc("""
+List users.
+
+SDK may hide the "maxpagesize" from API signature. The functionality of "maxpagesize" could be in related language Page model.
+
+Expected query parameter:
+skip=5
+maxpagesize=3
+
+Expected response body:
+```json
+{
+  "value":[
+    {
+      "name":"user5"
+    },
+    {
+      "name":"user6"
+    },
+    {
+      "name":"user7"
+    }
+  ],
+  "nextLink": "{endpoint}//payload/pageable?skip=8&maxpagesize=3"
+}
+```
+
+Expected query parameter:
+skip=8
+maxpagesize=3
+
+```json
+{
+  "value":[
+    {
+      "name":"user8"
+    }
+  ]
+}
+```
+""")
+@doc("List users")
+op list(...SkipQueryParameter, ...MaxPageSizeQueryParameter): Page<User>;
```

Comment:
I didn't use Azure.Core operation template.

I tried to write this case without model from Azure.Core, but Page there has further ref of `ResourceLocation` template.
https://github.com/Azure/typespec-azure/blob/ac3dc1566e3dea2e245f2e0034e42fcccbf6d2f9/packages/typespec-azure-core/lib/models.tsp#L14-L25